src/reducers/api.ts

```diff
@@ -0,0 +1,43 @@
+import {
+  FETCH_USERS,
+  FETCH_USERS_SUCCESS,
+  FETCH_USERS_ERROR
+} from "../actions/api";
+
+import { IApiState } from "../types";
+
+export const defaultState: IApiState = {
+  users: [],
+  isLoading: false,
+  error: null
+};
+
+const apiState = (state = defaultState, action: any): IApiState => {
+  switch (action.type) {
+    case FETCH_USERS:
+      return {
+        ...state,
+        isLoading: true
+      };
+    case FETCH_USERS_SUCCESS:
+      return {
+        ...state,
+        users: action.users,
+        isLoading: false
+      };
+    case FETCH_USERS_ERROR:
+      return {
+        ...state,
+        users: [],
+        isLoading: false,
+        error: action.error
+      };
+
+    default:
+      return state;
+  }
+};
+
+export default {
+  apiState
```

Comment:
I'm curious as to why this is wrapped in an object literal?

Reply:
apiState is the attribute I want to end up on the final global store. Exporting as an object like this allows me to use the spread operator in the rootReducer, applying the apiState property to the global store. Semantically, it makes more sense to me to spread the reducers properties when combining them in rootReducer, so I export this as an object literal. Just my way, certainly can be talked about though.